tests/unit/libstore-support/tests/nix_api_store.hh

```diff
@@ -24,21 +26,35 @@ public:
     {
         nix_store_free(store);
 
-        for (auto & path : fs::recursive_directory_iterator(nixDir)) {
-            fs::permissions(path, fs::perms::owner_all);
-        }
-        fs::remove_all(nixDir);
+        remove_test_files();
     }
 
     Store * store;
     std::string nixDir;
     std::string nixStoreDir;
 
 protected:
+
+    void remove_test_files()
+    {
+        for (auto & path : fs::recursive_directory_iterator(nixDir)) {
+            fs::permissions(path, fs::perms::owner_all);
+        }
+        fs::remove_all(nixDir);
+    }
+
     void init_local_store()
     {
-        auto tmpl = nix::getEnv("TMPDIR").value_or("/tmp") + "/tests_nix-store.XXXXXX";
+#ifdef _WIN32
+        // no `mktempd` with MinGW
+        nixDir = nix::defaultTempDir() + "/tests_nix-store";
+        remove_test_files();
+        fs::create_directory(nixDir);
```

Comment:
That looks scarily racy :astonished: 
If there's no `mktemp` equivalent at all, maybe at least add the current pid or something as a suffix?

Reply:
Oh good point, thanks!

I made it try numbers sequentially, which is not the most efficient algorithm, but will avoid such races.

Reply:
Simple and brutal, love that :D